hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -997,6 +998,15 @@ private void addNotificationLog(NotificationEvent event, ListenerEvent listenerE
         stmt.execute("SET @@session.sql_mode=ANSI_QUOTES");
       }
 
+      // Derby doesn't allow FOR UPDATE to lock the row being selected (See https://db.apache
+      // .org/derby/docs/10.1/ref/rrefsqlj31783.html) . So lock the whole table. Since there's
+      // only one row in the table, this shouldn't cause any performance degradation. Also we not
+      // suggest Derby to be used in production so that's fine.
```

Comment:
You are right!. Removed.

I was misled because of https://cwiki.apache.org/confluence/display/Hive/AdminManual+Metastore+Administration#AdminManualMetastoreAdministration-Local/EmbeddedMetastoreDatabase(Derby), where it says it's only for unit tests. But that's only for embedded db. Sorry for the confusion